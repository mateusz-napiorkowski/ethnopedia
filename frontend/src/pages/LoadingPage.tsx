import { ReactComponent as LoadingWheel } from "../assets/icons/loadingWheel.svg"

const LoadingPage = () => {
    return <div role="status" className="flex justify-center items-center h-screen bg-gray-50 dark:bg-gray-900">
        <LoadingWheel />
    </div>
}
export default LoadingPage