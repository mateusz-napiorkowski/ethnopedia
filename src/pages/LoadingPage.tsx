import { ReactComponent as LoadingWheel } from "../assets/icons/loadingWheel.svg"

const LoadingPage = () => {
    return <div role="status">
        <LoadingWheel />
    </div>
}
export default LoadingPage