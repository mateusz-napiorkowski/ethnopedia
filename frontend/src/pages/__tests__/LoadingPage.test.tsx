import '@testing-library/jest-dom';
import { render } from '@testing-library/react'
import LoadingPage from "../LoadingPage"
import { MemoryRouter, Routes, Route } from "react-router-dom";

const renderComponent = () => {
    return render(
        <MemoryRouter initialEntries={[`/`]}>
            <Routes>
                <Route path="/" element={<LoadingPage/>}/>
            </Routes>  
        </MemoryRouter>
    );
};

describe("Loading page tests", () => {
    beforeEach(() => {
        jest.clearAllMocks()
    });

    it("should render loading page", () => {           
        const {container} = renderComponent()
        
        expect(container).toMatchSnapshot()
    })
})