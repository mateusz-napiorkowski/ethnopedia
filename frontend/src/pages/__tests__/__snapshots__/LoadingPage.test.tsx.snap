// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading page tests should render loading page 1`] = `
<div>
  <div
    class="flex justify-center items-center h-screen bg-gray-50 dark:bg-gray-900"
    role="status"
  >
    <svg />
  </div>
</div>
`;
